pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -564,9 +577,42 @@ func getPipelineRunResults(pipelineSpec *v1beta1.PipelineSpec, resolvedResultRef
 	return results
 }
 
+func getRunsStatus(pr *v1beta1.PipelineRun, state []*resources.ResolvedPipelineRunTask) map[string]*v1beta1.PipelineRunRunStatus {
+	status := map[string]*v1beta1.PipelineRunRunStatus{}
+	for _, rprt := range state {
+		if !rprt.IsCustomTask() {
+			continue
+		}
+		if rprt.Run == nil && rprt.ResolvedConditionChecks == nil {
+			continue
+		}
+
+		var prrs *v1beta1.PipelineRunRunStatus
+		if rprt.Run != nil {
+			prrs = pr.Status.Runs[rprt.RunName]
+		}
+
+		if prrs == nil {
+			prrs = &v1beta1.PipelineRunRunStatus{
+				PipelineTaskName: rprt.PipelineTask.Name,
+			}
+		}
+
+		if rprt.Run != nil {
+			// TODO(jasonhall): Support RunStatus.
+			//prrs.Status = &rprt.Run.Status
+		}
+		status[rprt.RunName] = prrs
```

Comment:
Add another TODO here to handle the status of the pipeline task's condition checks (the same way as `getTaskRunsStatus` does)?  Or are Condition checks not supported with custom tasks?

Reply:
Conditions should be supported, TODO added. Thanks!